ietf/sync/rfceditor.py

```diff
@@ -367,80 +370,140 @@ def update_docs_from_rfc_index(index_data, errata_data, skip_older_than_date=Non
+            # Create DocAlias (for consistency until we drop DocAlias altogether)
+            alias, _ = DocAlias.objects.get_or_create(name=doc.name)
             alias.docs.add(doc)
-            changes.append("created alias %s" % prettify_std_name(name))
+            rfc_changes.append(f"created alias {prettify_std_name(doc.name)}")
+
+        if draft_name:
+            try:
+                draft = Document.objects.get(name=draft_name)
+            except Document.DoesNotExist:
+                pass
+            else:
+                # Ensure the draft is in the "rfc" state and move its files to the archive
+                # if necessary.
+                if draft.get_state_slug() != "rfc":
+                    draft.set_state(
+                        State.objects.get(used=True, type="draft", slug="rfc")
+                    )
+                    move_draft_files_to_archive(draft, draft.rev)
+                    draft_changes.append(f"changed state to {draft.get_state()}")
+
+                # Ensure the draft and rfc are linked with a "became_rfc" relationship
+                r, created_relateddoc = RelatedDocument.objects.get_or_create(
+                    source=draft, target=doc, relationship_id="became_rfc"
+                )
+                if created_relateddoc:
+                    change = "created {rel_name} relationship between {pretty_draft_name} and {pretty_rfc_name}".format(
+                        rel_name=r.relationship.name.lower(),
+                        pretty_draft_name=prettify_std_name(draft),
+                        pretty_rfc_name=prettify_std_name(doc),
+                    )
+                    draft_changes.append(change)
+                    rfc_changes.append(change)
+
+                # Ensure draft is in the correct iesg and stream states
+                for t in (
+                    "draft-iesg",
+                    "draft-stream-iab",
+                    "draft-stream-irtf",
+                    "draft-stream-ise",
+                ):
+                    prev_state = draft.get_state(t)
+                    if prev_state is not None:
+                        if prev_state.slug not in ("pub", "idexists"):
+                            new_state = State.objects.select_related("type").get(
+                                used=True, type=t, slug="pub"
+                            )
+                            draft.set_state(new_state)
+                            draft_changes.append(
+                                f"changed {new_state.type.label} to {new_state}"
+                            )
+                            e = update_action_holders(draft, prev_state, new_state)
+                            if e:
+                                draft_events.append(e)
+                    elif t == "draft-iesg":
+                        draft.set_state(
+                            State.objects.get(type_id="draft-iesg", slug="idexists")
+                        )
```

Comment:
This doesn't make sense to me.
Instead, verify that the draft stream matches the rfc stream, and then verify the states only for that stream.

Reply:
I see below that this is what the old code did, but I think that was wrong.